src/main/java/com/epam/ta/reportportal/core/bts/handler/impl/UpdateBugTrackingSystemHandlerImpl.java

```diff
@@ -145,22 +134,55 @@ public OperationCompletionRS updateBugTrackingSystem(UpdateBugTrackingSystemRQ u
 	}
 
 	@Override
-	public OperationCompletionRS integrationConnect(UpdateBugTrackingSystemRQ updateRequest, Long integrationId,
+	public OperationCompletionRS integrationConnect(BtsConnectionTestRQ connectionTestRQ, Long integrationId,
 			ReportPortalUser.ProjectDetails projectDetails) {
-		Integration integration = integrationRepository.findById(integrationId)
-				.orElseThrow(() -> new ReportPortalException(INTEGRATION_NOT_FOUND, integrationId));
 
-		Integration details = new BugTrackingSystemBuilder().addUrl(updateRequest.getUrl())
-				.addBugTrackingProject(updateRequest.getProject())
-				.get();
+		Project project = projectRepository.findById(projectDetails.getProjectId())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.PROJECT_NOT_FOUND, projectDetails.getProjectName()));
 
-		Optional<BtsExtension> extension = pluginBox.getInstance(details.getType().getName(), BtsExtension.class);
+		Integration integration = getBugTrackingSystemHandler.getEnabledByProjectIdAndIdOrGlobalById(projectDetails, integrationId);
 
-		expect(extension, Optional::isPresent).verify(UNABLE_INTERACT_WITH_INTEGRATION, integration.getProject().getId());
-		expect(extension.get().connectionTest(details), equalTo(true)).verify(UNABLE_INTERACT_WITH_INTEGRATION,
-				integration.getProject().getId()
+		IntegrationParams connectionParams = createConnectionParams(ofNullable(integration.getParams()).orElseGet(IntegrationParams::new),
+				connectionTestRQ
+		);
+		Integration connectionIntegration = new Integration();
+		connectionIntegration.setParams(connectionParams);
+		connectionIntegration.setType(integration.getType());
+		connectionIntegration.setProject(project);
+		connectionIntegration.setEnabled(integration.isEnabled());
+
+		Optional<BtsExtension> extension = pluginBox.getInstance(connectionIntegration.getType().getName(), BtsExtension.class);
+
+		expect(extension, Optional::isPresent).verify(UNABLE_INTERACT_WITH_INTEGRATION,
+				Suppliers.formattedSupplier("Plugin extension for integration with type = '{}' has not been found",
+						connectionIntegration.getType().getName()
+				)
+		);
+		expect(extension.get().connectionTest(connectionIntegration), equalTo(true)).verify(UNABLE_INTERACT_WITH_INTEGRATION,
+				Suppliers.formattedSupplier("Connection to the integration with id = '{}' refused", integrationId)
 		);
 
-		return new OperationCompletionRS("Connection to ExternalSystem with ID = '" + integrationId + "' is successfully performed.");
+		return new OperationCompletionRS("Connection to  BTS Integration with ID = '" + integrationId + "' is successfully performed.");
+	}
+
+	private IntegrationParams createConnectionParams(IntegrationParams paramsToUpdate, BtsConnectionTestRQ connectionTestRQ) {
```

Comment:
maybe it's better to call it "prepareConnectionTestingParams(IntegrationParams existedParams, String url, String project) ?

Reply:
cause as i understand you are not updating integration params, you just specifying params for test connection

Reply:
renamed